### PR TITLE
Update terraform getting started doc to use identity file.

### PR DESCRIPTION
**Pull request by Joerger**
Identity file is now the suggested credential loader for terraform because it can connect through a Teleport Proxy.

Fixes https://github.com/gravitational/teleport/issues/6566

### CODE REVIEW

File: terraform/GETTING_STARTED.md [hunk 1]
```diff
@@ -69,10 +69,8 @@ terraform {
 }
 
 provider "teleport" {
-  addr         = "localhost:3025"
-  cert_path    = "tf.crt"
-  key_path     = "tf.key"
-  root_ca_path = "tf.cas"
+  addr               = "localhost:3025"
```
Comment (benarent) [inline terraform/GETTING_STARTED.md:72]:
```suggestion
  # Update addr to point to Teleport Auth/Proxy
  # e.g. addr = "example.teleport.sh:3025"
  addr               = "localhost:3025"
```

File: terraform/GETTING_STARTED.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ Run:
 
 ```
 tctl create terraform.yaml
-tctl auth sign --format=tls --user=terraform --out=tf --ttl=10h
+tctl auth sign --format=file --user=terraform --out=terraform-identity --ttl=10h
 ```
```
Comment (benarent) [inline terraform/GETTING_STARTED.md:56]: Tip: Make sure the user can impersonate the new terraform role and user. 

```
spec:
  allow:
    impersonate:
      roles:
      - terraform
      users:
      - terraform
```

Comment (Joerger) [reply terraform/GETTING_STARTED.md:56]: I'm not sure I understand. Why does the Terraform user/role need to impersonate itself?

Comment (benarent) [reply terraform/GETTING_STARTED.md:56]: > Why does the Terraform user/role need to impersonate itself?

I had to add it my role, `access`, meaning that the `access` role can impersonate the user. This is important for cloud users. As they'll want to impersonate as the user and not login and obtain the certs as that user.

**Review by klizhentas — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**